### PR TITLE
feat: add npm package to homestar runtime

### PR DESCRIPTION
**Pull request by hugomrdias**
# Description

adds npm packages to homestar-runtime.
CI is not integrated yet, there only manual instructions in the readme.

## Type of change

- [x] New feature (non-breaking change that adds functionality)

### CODE REVIEW

File: homestar-runtime/npm/readme.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+rustup target add aarch64-unknown-linux-gnu
+rustup target add x86_64-unknown-linux-musl
+
+
+export node_version=0.0.1
+export bin="homestar"
+
+
+## darwin arm64
+cargo build -p homestar-runtime --locked --release --target aarch64-apple-darwin
```
Comment (zeeshanlakhani) [inline homestar-runtime/npm/readme.md:16]: --features ansi-logs

File: homestar-runtime/npm/readme.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+# publish the package
+cd "${node_pkg}"
+npm publish --access public
+```
+
+## TODO
+
+- [ ] move this to CI
+- [ ] add windows
+- [ ] --features ansi-logs
```
Comment (zeeshanlakhani) [inline homestar-runtime/npm/readme.md:60]: 👍🏽

**Review by zeeshanlakhani — COMMENTED** [review]
So far so good. I can help w/ the CI tmrw for non-windows first.

File: homestar-runtime/npm/readme.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+cross build -p homestar-runtime --features ansi-logs --locked --release --target aarch64-unknown-linux-gnu
+export node_os=linux
+export node_arch=arm64
+export node_pkg="${bin}-${node_os}-${node_arch}"
+mkdir -p "binaries/${node_pkg}/bin"
+envsubst < package.json.tmpl > "binaries/${node_pkg}/package.json"
+cp "../../target/aarch64-unknown-linux-gnu/release/${bin}" "binaries/${node_pkg}/bin"
+
+## linux x64
+cross build -p homestar-runtime --features ansi-logs --locked --release --target x86_64-unknown-linux-musl
```
Comment (zeeshanlakhani) [inline homestar-runtime/npm/readme.md:43]: fyi, ansi-logs will be horrible for grafana's. If we think of this more for dev, then it's good to have it on :).

File: .github/workflows/builds.yml [hunk 4]
```diff
@@ -15,12 +15,12 @@ on:
...
   # for debugging
-  # pull_request:
-  #   branches: [ '**' ]
+  pull_request:
+    branches: [ '**' ]
 
-concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
+# concurrency:
```
Comment (zeeshanlakhani) [inline .github/workflows/builds.yml:21]: for debugging?

File: .github/workflows/builds.yml [hunk 5]
```diff
@@ -95,6 +100,72 @@ jobs:
...
+      - run: echo ${{  github.event_name }}
+      - name: Publish production
+        if: github.event_name == 'release' && github.event.action == 'published'
+        run: |
+          cd "homestar-runtime/npm/${{ env.node_pkg }}"
+          pnpm -r publish --access=public
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+      - name: Publish RC
+        if: github.event_name == 'workflow_dispatch' || github.event_name == 'pull_request'
```
Comment (zeeshanlakhani) [inline .github/workflows/builds.yml:159]: did we want to do this on PR or merge? Now, post-workshop, we'll be merging consistently into `main`, as PR's could still have broken code?

Comment (hugomrdias) [reply .github/workflows/builds.yml:159]: just testing for now, its will be just `workflow_dispatch` for RC and the production one

Comment (zeeshanlakhani) [reply .github/workflows/builds.yml:159]: Ok, makes sense.

File: .github/workflows/builds.yml (hunk 4 above)
Comment (zeeshanlakhani) [inline .github/workflows/builds.yml:21]: note: bring back before merge.

File: .github/workflows/builds.yml [hunk 6]
```diff
@@ -95,6 +100,72 @@ jobs:
...
+        run: |
+          cd "homestar-runtime/npm/${{ env.node_pkg }}"
+          ls -R
+          npm version $(cargo get package.version)-rc.$(date +%s) --git-tag-version false
+          npm publish --access public --tag rc
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+
```
Comment (zeeshanlakhani) [inline .github/workflows/builds.yml:166]: minor: spacing

File: homestar-runtime/npm/readme.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+envsubst < package.json.tmpl > "binaries/${node_pkg}/package.json"
+cp "../../target/x86_64-unknown-linux-musl/release/${bin}" "binaries/${node_pkg}/bin"
+
+# publish the package
+cd "${node_pkg}"
+npm version $(cargo get package.version)-rc.$(date +%s) --git-tag-version false
+npm publish --access public --tag rc
+```
+
+## TODO
```
Comment (zeeshanlakhani) [inline homestar-runtime/npm/readme.md:59]: note: remove todos

File: .github/workflows/builds.yml [hunk 8]
```diff
@@ -16,7 +16,7 @@ on:
 
   # for debugging
```
Comment (zeeshanlakhani) [inline .github/workflows/builds.yml:17]: minor: @hugomrdias since you're in this file, can you remove:

`# TODO: bring back release hook (vs on PR)`

**Review by zeeshanlakhani — APPROVED** [review]
one, super minor ask :).

**Comment by hugomrdias** [comment]
done

**Pull request merged**